batch/batch/driver/instance_collection.py

```diff
@@ -123,6 +123,11 @@ async def check_on_instance(self, instance):
                 return
             raise
 
+        if instance.failed_request_count > 10 and time_msecs() - instance.last_updated > 5 * 60 * 1000:
+            log.exception(f'removing {instance} with {instance.failed_request_count} failed request counts after more than 5 minutes')
+            await self.remove_instance(instance, 'not_responding')
```

Comment:
shouldn't we delete it too?

Reply:
I feel like what we need to do depends on the instance state no? If the instance is stopping or terminated, we just need to deactivate.

Provisioning instances will get killed anyway.

This really seems only relevant to STAGING or RUNNING instances, right? It sounds like the if on 153 should be if pending _or_ many failed requests.

Reply:
I made it so the check only happens for active instances. Good catch on the delete.